File: Anti-Flag.cpp
#include <iostream>
#include <string>
#include <vector>
#include <filesystem>
#include <thread>
#include <chrono>
#include <Windows.h>

namespace fs = std::filesystem;

bool deleteFileOrDirectory(const fs::path& path) {
    try {
        if (fs::exists(path)) {
            if (fs::is_regular_file(path)) {
                fs::remove(path);
            } else if (fs::is_directory(path)) {
                fs::remove_all(path);
            }
            return true;
        }
    } catch (const std::exception& e) {
        std::cerr << "Error deleting " << path << ": " << e.what() << std::endl;
    }
    return false;
}

std::string readRegistryValue(HKEY key, const std::wstring& subkey, const std::wstring& valueName) {
    HKEY subkeyHandle;
    DWORD dataSize;
    DWORD dataType;
    if (RegOpenKeyExW(key, subkey.c_str(), 0, KEY_READ, &subkeyHandle) == ERROR_SUCCESS) {
        if (RegQueryValueExW(subkeyHandle, valueName.c_str(), NULL, &dataType, NULL, &dataSize) == ERROR_SUCCESS) {
            if (dataType == REG_SZ) {
                std::vector<wchar_t> data(dataSize / sizeof(wchar_t));
                if (RegQueryValueExW(subkeyHandle, valueName.c_str(), NULL, &dataType, reinterpret_cast<LPBYTE>(data.data()), &dataSize) == ERROR_SUCCESS) {
                    RegCloseKey(subkeyHandle);
                    return std::wstring(data.begin(), data.end() - 1); // Remove null terminator
                }
            }
        }
        RegCloseKey(subkeyHandle);
    }
    return "";
}

void deleteRegistryKey(HKEY key, const std::wstring& subkey) {
    HKEY subkeyHandle;
    if (RegOpenKeyExW(key, subkey.c_str(), 0, KEY_ALL_ACCESS, &subkeyHandle) == ERROR_SUCCESS) {
        RegDeleteTreeW(subkeyHandle, nullptr);
        RegCloseKey(subkeyHandle);
    }
}

void patchOverwatch() {
    const std::wstring userProfile = std::wstring(_wgetenv(L"USERPROFILE"));
    const std::wstring appData = std::wstring(_wgetenv(L"APPDATA"));
    const std::wstring programData = std::wstring(_wgetenv(L"ProgramData"));

    std::vector<fs::path> overwatchPaths = {
        fs::path(userProfile) / L"AppData\\Local\\Battle.net",
        fs::path(userProfile) / L"AppData\\Local\\Blizzard",
        fs::path(userProfile) / L"AppData\\Local\\Blizzard Entertainment",
        fs::path(userProfile) / L"AppData\\Roaming\\Battle.net",
        fs::path(userProfile) / L"Documents\\Overwatch\\Logs",
        fs::path(programData) / L"Battle.net\\Setup",
        fs::path(programData) / L"Battle.net\\Agent\\data",
        fs::path(programData) / L"Battle.net\\Agent\\Logs",
        fs::path(programData) / L"Blizzard Entertainment"
    };

    std::wcout << L"Deleting Overwatch-related paths..." << std::endl;

    for (const auto& path : overwatchPaths) {
        deleteFileOrDirectory(path);
    }

    const std::wstring installLocation = readRegistryValue(HKEY_LOCAL_MACHINE, L"SOFTWARE\\WOW6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\Overwatch", L"InstallLocation");

    if (!installLocation.empty()) {
        std::vector<fs::path> cacheDirs = {
            fs::path(installLocation) / L"_retail_\\cache",
            fs::path(installLocation) / L"_retail_\\GPUCache"
        };

        std::wcout << L"Deleting Overwatch cache directories..." << std::endl;

        for (const auto& cacheDir : cacheDirs) {
            deleteFileOrDirectory(cacheDir);
        }
    }

    const fs::path agentsPath = fs::path(programData) / L"Battle.net\\Agent";
    if (fs::exists(agentsPath)) {
        std::vector<fs::path> agents;
        for (const auto& entry : fs::directory_iterator(agentsPath)) {
            if (entry.path().stem().string().starts_with("Agent")) {
                agents.push_back(entry.path());
            }
        }

        if (!agents.empty()) {
            const auto latestAgent = *std::max_element(agents.begin(), agents.end(), [](const fs::path& p1, const fs::path& p2) {
                return fs::last_write_time(p1) < fs::last_write_time(p2);
            });

            for (const auto& agent : agents) {
                if (agent != latestAgent) {
                    deleteFileOrDirectory(agent);
                }
            }
        }
    }

    const std::vector<std::wstring> registryKeys = {
        L"SOFTWARE\\WOW6432Node\\Blizzard Entertainment",
        L"SOFTWARE\\Blizzard Entertainment",
        L"SOFTWARE\\Activision",
        L"HKEY_CLASSES_ROOT\\Applications\\Overwatch.exe",
        L"HKEY_CURRENT_USER\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\CapabilityAccessManager\\ConsentStore\\microphone\\NonPackaged\\C:#Program Files (x86)#Overwatch#_retail_#Overwatch.exe",
        L"HKEY_CURRENT_USER\\SOFTWARE\\Microsoft\\RADAR\\HeapLeakDetection\\DiagnosedApplications\\Overwatch.exe",
        L"HKEY_CURRENT_USER\\VirtualStore\\MACHINE\\SOFTWARE\\WOW6432Node\\Activision",
        L"HKEY_CURRENT_USER\\SOFTWARE\\Classes\\VirtualStore\\MACHINE\\SOFTWARE\\WOW6432Node\\Activision"
    };

    std::wcout << L"Deleting Overwatch registry keys..." << std::endl;

    for (const auto& key : registryKeys) {
        const std::wstring baseKey = key.substr(0, key.find('\\'));
        const std::wstring subkey = key.substr(key.find('\\') + 1);

        if (baseKey == L"HKEY_CLASSES_ROOT") {
            deleteRegistryKey(HKEY_CLASSES_ROOT, subkey);
        } else if (baseKey == L"HKEY_CURRENT_USER") {
            deleteRegistryKey(HKEY_CURRENT_USER, subkey);
        } else if (baseKey == L"HKEY_LOCAL_MACHINE") {
            deleteRegistryKey(HKEY_LOCAL_MACHINE, subkey);
        } else if (baseKey == L"HKEY_USERS") {
            deleteRegistryKey(HKEY_USERS, subkey);
        } else if (baseKey == L"HKEY_CURRENT_CONFIG") {
            deleteRegistryKey(HKEY_CURRENT_CONFIG, subkey);
        }
    }
}

void patchCookies() {
    const std::wstring userProfile = std::wstring(_wgetenv(L"USERPROFILE"));
    const std::wstring appData = std::wstring(_wgetenv(L"APPDATA"));

    std::vector<fs::path> browserCookiePaths = {
        fs::path(userProfile) / L"AppData\\Local\\BraveSoftware\\Brave-Browser\\User Data\\Default\\Cookies",
        fs::path(appData) / L"Google\\Chrome\\User Data\\Default\\Cookies",
        fs::path(appData) / L"Opera Software\\Opera Stable\\Cookies"
    };

    std::wcout << L"Deleting browser cookie files..." << std::endl;

    for (const auto& path : browserCookiePaths) {
        deleteFileOrDirectory(path);
    }

    const fs::path firefoxProfilesPath = fs::path(appData) / L"Mozilla\\Firefox\\Profiles";
    if (fs::exists(firefoxProfilesPath)) {
        for (const auto& entry : fs::directory_iterator(firefoxProfilesPath)) {
            const fs::path cookiesPath = entry.path() / L"cookies.sqlite";
            deleteFileOrDirectory(cookiesPath);
        }
    }
}

void flushNetworkAndExit() {
    std::cout << "Flushing network..." << std::endl;
    system("ipconfig /flushdns"); // Flush DNS cache
    std::cout << "Network flushed." << std::endl;
    std::exit(0);
}

int main() {
    std::cout << "Starting patching process..." << std::endl;

    std::thread overwatchThread(patchOverwatch);
    std::thread cookiesThread(patchCookies);

    overwatchThread.join();
    cookiesThread.join();

    std::cout << "Patching completed successfully." << std::endl;

    flushNetworkAndExit();

    return 0;
}

File: Anti-Flag.rs
use std::env;
use std::fs;
use std::io;
use std::path::Path;
use std::thread;
use std::time::Duration;
use colored::*; // You need to add this crate: `colored = "2"`
use winreg::enums::*;
use winreg::RegKey;

fn main() {
    println!("{}", "Starting patching process...".green());

    let handle = thread::spawn(|| patch_overwatch());
    let handle2 = thread::spawn(|| patch_cookies());

    handle.join().unwrap();
    handle2.join().unwrap();

    println!("{}", "Patching completed successfully.".green());

    flush_network_and_exit();
}

fn patch_overwatch() {
    let user_profile = env::var("USERPROFILE").unwrap();
    let app_data = env::var("APPDATA").unwrap();

    let overwatch_paths = vec![
        Path::new(&user_profile)
            .join("AppData")
            .join("Local")
            .join("Battle.net"),
        Path::new(&user_profile)
            .join("AppData")
            .join("Local")
            .join("Blizzard"),
        Path::new(&user_profile)
            .join("AppData")
            .join("Local")
            .join("Blizzard Entertainment"),
        Path::new(&user_profile)
            .join("AppData")
            .join("Roaming")
            .join("Battle.net"),
        Path::new(&user_profile)
            .join("Documents")
            .join("Overwatch")
            .join("Logs"),
        Path::new(env::var("ProgramData").unwrap())
            .join("Battle.net")
            .join("Setup"),
        Path::new(env::var("ProgramData").unwrap())
            .join("Battle.net")
            .join("Agent")
            .join("data"),
        Path::new(env::var("ProgramData").unwrap())
            .join("Battle.net")
            .join("Agent")
            .join("Logs"),
        Path::new(env::var("ProgramData").unwrap())
            .join("Blizzard Entertainment"),
    ];

    println!("{}", "Deleting Overwatch-related paths...".yellow());

    for path in &overwatch_paths {
        delete_file_or_dir(path);
    }

    let install_location = read_registry_value(
        RegKey::predef(HKEY_LOCAL_MACHINE),
        r"SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\Overwatch",
        "InstallLocation",
    );

    if let Ok(install_location) = install_location {
        if !install_location.is_empty() {
            let cache_dirs = vec![
                Path::new(&install_location)
                    .join("_retail_")
                    .join("cache"),
                Path::new(&install_location)
                    .join("_retail_")
                    .join("GPUCache"),
            ];

            println!("{}", "Deleting Overwatch cache directories...".yellow());

            for cache_dir in &cache_dirs {
                delete_file_or_dir(cache_dir);
            }
        }
    }

    let agents_path = Path::new(env::var("ProgramData").unwrap())
        .join("Battle.net")
        .join("Agent");

    if agents_path.exists() {
        let agents = fs::read_dir(&agents_path)
            .unwrap()
            .filter_map(|entry| entry.ok())
            .map(|entry| entry.path())
            .filter(|path| path.starts_with(&agents_path.join("Agent")))
            .collect::<Vec<_>>();

        if !agents.is_empty() {
            let latest_agent = agents
                .iter()
                .max_by_key(|path| path.metadata().unwrap().created().unwrap())
                .unwrap();

            for agent in agents {
                if agent != latest_agent {
                    delete_file_or_dir(&agent);
                }
            }
        }
    }

    let registry_keys = vec![
        r"SOFTWARE\WOW6432Node\Blizzard Entertainment",
        r"SOFTWARE\Blizzard Entertainment",
        r"SOFTWARE\Activision",
        r"HKEY_CLASSES_ROOT\Applications\Overwatch.exe",
        r"HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\CapabilityAccessManager\ConsentStore\microphone\NonPackaged\C:#Program Files (x86)#Overwatch#_retail_#Overwatch.exe",
        r"HKEY_CURRENT_USER\SOFTWARE\Microsoft\RADAR\HeapLeakDetection\DiagnosedApplications\Overwatch.exe",
        r"HKEY_CURRENT_USER\VirtualStore\MACHINE\SOFTWARE\WOW6432Node\Activision",
        r"HKEY_CURRENT_USER\SOFTWARE\Classes\VirtualStore\MACHINE\SOFTWARE\WOW6432Node\Activision",
    ];

    println!("{}", "Deleting Overwatch registry keys...".yellow());

    for key in &registry_keys {
        delete_registry_key(key);
    }
}

fn patch_cookies() {
    let user_profile = env::var("USERPROFILE").unwrap();
    let app_data = env::var("APPDATA").unwrap();

    let browser_cookie_paths = vec![
        Path::new(&user_profile)
            .join("AppData")
            .join("Local")
            .join("BraveSoftware")
            .join("Brave-Browser")
            .join("User Data")
            .join("Default")
            .join("Cookies"),
        Path::new(&app_data)
            .join("Google")
            .join("Chrome")
            .join("User Data")
            .join("Default")
            .join("Cookies"),
        Path::new(&app_data)
            .join("Opera Software")
            .join("Opera Stable")
            .join("Cookies"),
    ];

    println!("{}", "Deleting browser cookie files...".yellow());

    for path in &browser_cookie_paths {
        delete_file_or_dir(path);
    }

    let firefox_profiles_path = Path::new(&app_data).join("Mozilla").join("Firefox").join("Profiles");

    if firefox_profiles_path.exists() {
        for profile_dir in fs::read_dir(&firefox_profiles_path).unwrap() {
            if let Ok(profile_dir) = profile_dir {
                let cookies_path = profile_dir.path().join("cookies.sqlite");
                delete_file_or_dir(&cookies_path);
            }
        }
    }
}

fn delete_file_or_dir<P: AsRef<Path>>(path: P) {
    let path = path.as_ref();
    if path.exists() {
        if path.is_file() {
            if let Err(e) = fs::remove_file(path) {
                println!(
                    "{}",
                    format!("Error deleting {}: {}", path.display(), e).red()
                );
            }
        } else if path.is_dir() {
            if let Err(e) = fs::remove_dir_all(path) {
                println!(
                    "{}",
                    format!("Error deleting {}: {}", path.display(), e).red()
                );
            }
        }
    }
}

fn read_registry_value(key: RegKey, subkey: &str, value_name: &str) -> io::Result<String> {
    let key = key.open_subkey_with_flags(subkey, KEY_READ)?;
    let value: String = key.get_value(value_name)?;
    Ok(value)
}

fn delete_registry_key(key_path: &str) {
    let parts: Vec<&str> = key_path.split('\\').collect();
    if parts.len() < 2 {
        return;
    }

    let base_key = match parts[0] {
        "HKEY_CLASSES_ROOT" => RegKey::predef(HKEY_CLASSES_ROOT),
        "HKEY_CURRENT_USER" => RegKey::predef(HKEY_CURRENT_USER),
        "HKEY_LOCAL_MACHINE" => RegKey::predef(HKEY_LOCAL_MACHINE),
        "HKEY_USERS" => RegKey::predef(HKEY_USERS),
        "HKEY_CURRENT_CONFIG" => RegKey::predef(HKEY_CURRENT_CONFIG),
        _ => {
            println!("{}", format!("Invalid base key name: {}", parts[0]).red());
            return;
        }
    };

    let subkey_path = parts[1..].join("\\");

    if let Ok(key) = base_key.open_subkey_with_flags(&subkey_path, KEY_ALL_ACCESS) {
        if let Err(e) = key.delete_subkey_all() {
            println!(
                "{}",
                format!("Error deleting registry key: {}", e).red()
            );
        }
    }
}

fn flush_network_and_exit() {
    println!("{}", "Network flushed.".green());

    std::process::exit(0);
}
